qiskit_experiments/base_experiment.py

```diff
@@ -201,59 +189,106 @@ def circuits(
+                " as it is determined by the experiment physical qubits."
+            )
+        self._transpile_options.update_options(**fields)
 
-        .. note::
-            These circuits should be on qubits ``[0, .., N-1]`` for an
-            *N*-qubit experiment. The circuits mapped to physical qubits
-            are obtained via the :meth:`transpiled_circuits` method.
+    @classmethod
+    def _default_run_options(cls) -> Options:
+        """Default options values for the experiment :meth:`run` method."""
+        return Options()
+
+    @property
+    def run_options(self) -> Options:
+        """Return options values for the experiment :meth:`run` method."""
+        return self._run_options
+
+    def set_run_options(self, **fields):
+        """Set options values for the experiment  :meth:`run` method.
+
+        Args:
+            fields: The fields to update the options
         """
-        # Filter kwargs to circuit and transpile options
-        circuit_options = {}
-        transpile_options = self.__transpile_defaults__.copy()
-        for key, value in kwargs.items():
-            valid_key = False
-            if key in self._circuit_options:
-                circuit_options[key] = value
-                valid_key = True
-            if key in _TRANSPILE_OPTIONS:
-                transpile_options[key] = value
-                valid_key = True
-            if not valid_key:
-                raise QiskitError(
-                    f"{key} is not a valid kwarg for" f" {self.circuits} or {transpile}"
-                )
+        self._run_options.update_options(**fields)
+
+    @classmethod
+    def _default_analysis_options(cls) -> Options:
+        """Default options for analysis of experiment results."""
+        # Experiment subclasses can override this method if they need
+        # to set specific analysis options defaults that are different
+        # from the Analysis subclass `_default_options` values.
+        if cls.__analysis_class__:
+            return cls.__analysis_class__._default_options()
+        return Options()
 
-        # Generate circuits
-        circuits = self.circuits(backend=backend, **circuit_options)
+    @property
+    def analysis_options(self) -> Options:
+        """Return the analysis options for :meth:`run` analysis."""
+        return self._analysis_options
 
-        # Transpile circuits
-        if "initial_layout" in transpile_options:
-            raise QiskitError("Initial layout must be specified by the Experiement.")
-        transpile_options["initial_layout"] = self.physical_qubits
-        circuits = transpile(circuits, backend=backend, **transpile_options)
+    def set_analysis_options(self, **fields):
+        """Set the analysis options for :meth:`run` method.
 
-        return circuits
+        Args:
+            fields: The fields to update the options
+        """
+        self._analysis_options.update_options(**fields)
```

Comment:
This can have field validation as experiment options. Analysis class can define the list of valid options.

Reply:
Perhaps this is not feasible due to this [**kwargs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html#scipy.optimize.curve_fit).